lobby-db-dao/src/test/java/org/triplea/lobby/server/db/dao/UserJdbiDaoTest.java

```diff
@@ -0,0 +1,43 @@
+package org.triplea.lobby.server.db.dao;
+
+import static com.github.npathai.hamcrestopt.OptionalMatchers.isEmpty;
+import static com.github.npathai.hamcrestopt.OptionalMatchers.isPresentAndIs;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.junit5.DBUnitExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.triplea.lobby.server.db.JdbiDatabase;
+import org.triplea.test.common.Integration;
+
+@Integration
+@ExtendWith(DBUnitExtension.class)
+@DataSet(cleanBefore = true, value = "user/select.yml")
+class UserJdbiDaoTest {
+
+  private static final String USERNAME = "user";
+  private static final String PASSWORD =
+      "$2a$56789_123456789_123456789_123456789_123456789_123456789_";
+
+  private final UserJdbiDao userDao = JdbiDatabase.newConnection().onDemand(UserJdbiDao.class);
+
+  @Test
+  void lookupUserIdByName() {
+    assertThat(userDao.lookupUserIdByName("DNE"), isEmpty());
+    assertThat(userDao.lookupUserIdByName(USERNAME), isPresentAndIs(900000));
+  }
+
+  @Test
+  void getPassword() {
+    assertThat(userDao.getPassword(USERNAME), isPresentAndIs(PASSWORD));
+    assertThat(userDao.getPassword("DNE"), isEmpty());
+  }
+
+  @Test
+  void updateLastLogin() {
+    assertThat(userDao.updateLastLoginTime(USERNAME), is(1));
+    assertThat(userDao.updateLastLoginTime("DNE"), is(0));
```

Comment:
Wouldn't it make sense to check  if this actually updates the time here?
I know this isn't 100% trivial because we can't mock the DB now() function, but at least checking if the new date is more recent would be a win IMO

Reply:
Indeed this would be normally verified with an expected dataset. You're right about the 'now' function. One problem is there are no queries against that column, reading it would require a test-only DB command. It's possible, but we'd add complication in the test and production coupling to the test-code.

The other alternative would be to parameterize the date function that is passed in so we do not use the DB 'now()'. It's really ideal to use 'now()' as then the DB can take care of the date, no time zone or formatting issue, and the java method has fewer parameters.

I don't know if either of those two choices are really worth it for what is otherwise a pretty simple DB update. To some extent knowing we can execute the query against a live-DB is one of the biggest wins we get and would catch most problems we could have. For example if the DB column name had a typo, or the table, some of the more likely problems with a simple query where there is a refactor of table or columns names, we would catch those.

Of some note, the return value of the method was updated to be an 'int' and not a 'void' method call in order to enable this level testing so we are not *just* invoking the DB query.